src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -140,7 +141,7 @@ public FileSystemInfo ToFileSystemInfo()
         /// Returns the full path for find results, based on the initially provided path.
         /// </summary>
         public string ToSpecifiedFullPath() =>
-            PathHelpers.CombineNoChecks(OriginalRootDirectory, Directory.Slice(RootDirectory.Length), FileName);
+            Path.Join(OriginalRootDirectory, Directory.Slice(RootDirectory.Length), FileName);
```

Comment:
Is it guaranteed that `Directory.Length >= RootDirectory.Length`?

Reply:
Yes, as it is a series of name Joins starting with RootDirectory.